src/main/java/hexgui/htp/AnalyzeCommand.java

```diff
@@ -0,0 +1,220 @@
+// AnalyzeCommand.java
+
+package hexgui.htp;
+
+import static hexgui.hex.HexColor.BLACK;
+import static hexgui.hex.HexColor.WHITE;
+
+import hexgui.hex.ConstPointList;
+import hexgui.hex.HexColor;
+import hexgui.hex.HexPoint;
+import hexgui.hex.PointList;
+import java.io.File;
+
+/**
+ * Concrete analyze command including data for wildcard replacements. See GoGui documentation,
+ * chapter "Analyze Commands"
+ */
+public class AnalyzeCommand {
+  public AnalyzeCommand(AnalyzeDefinition definition) {
+    m_definition = definition;
+  }
+
+  public String getLabel() {
+    return m_definition.getLabel();
+  }
+
+  public String getCommand() {
+    return m_definition.getCommand();
+  }
+
+  public HexColor getColorArg() {
+    return m_colorArg;
+  }
+
+  public HexPoint getPointArg() {
+    return m_pointArg;
+  }
+
+  public PointList getPointListArg() {
+    return m_pointListArg;
+  }
+
+  public AnalyzeType getType() {
+    return m_definition.getType();
+  }
+
+  public String getResultTitle() {
+    StringBuilder buffer = new StringBuilder(getLabel());
+    if (needsColorArg() && m_colorArg != null) {
+      if (m_colorArg == BLACK) buffer.append(" Black");
+      else {
+        assert m_colorArg == WHITE;
+        buffer.append(" White");
+      }
+    }
+    if (needsPointArg() && m_pointArg != null) {
+      buffer.append(' ');
+      buffer.append(m_pointArg);
+    } else if (needsPointListArg()) {
+      for (HexPoint p : m_pointListArg) {
+        buffer.append(' ');
+        buffer.append(p);
+      }
+    }
+    if (needsStringArg() && m_stringArg != null) {
+      buffer.append(' ');
+      buffer.append(m_stringArg);
+    }
+    return buffer.toString();
+  }
+
+  public boolean isPointArgMissing() {
+    if (needsPointArg()) return (m_pointArg == null);
+    if (needsPointListArg()) return m_pointListArg.isEmpty();
+    return false;
+  }
+
+  public boolean isTextType() {
+    return m_definition.isTextType();
+  }
+
+  public boolean needsColorArg() {
+    return m_definition.needsColorArg();
+  }
+
+  public boolean needsFileArg() {
+    return m_definition.needsFileArg();
+  }
+
+  public boolean needsFileOpenArg() {
+    return m_definition.needsFileOpenArg();
+  }
+
+  public boolean needsFileSaveArg() {
+    return m_definition.needsFileSaveArg();
+  }
+
+  public boolean needsOnlyPointArg() {
+    return m_definition.needsOnlyPointArg();
+  }
+
+  public boolean needsOnlyPointAndColorArg() {
+    return m_definition.needsOnlyPointAndColorArg();
+  }
+
+  public boolean needsPointArg() {
+    return m_definition.needsPointArg();
+  }
+
+  public boolean needsPointListArg() {
+    return m_definition.needsPointListArg();
+  }
+
+  public boolean needsStringArg() {
+    return m_definition.needsStringArg();
+  }
+
+  public boolean needsOptStringArg() {
+    return m_definition.needsOptStringArg();
+  }
+
+  public String replaceWildCards(HexColor toMove) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 18. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 143-162 into a separate method.